runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -64,6 +75,8 @@
             .expireAfterAccess(runtime.getParameters().getCacheExpiryTime(), TimeUnit.SECONDS)
             .expireAfterWrite(runtime.getParameters().getCacheExpiryTime(), TimeUnit.SECONDS)
             .recordStats()
+            .removalListener((RemovalListener<Long, ILogData>) notification ->
```

Comment:
Just curious, would the historical give us some interesting info? for instance realize that we're replacing an entry over and over again or is this not aligned with what you're trying to collect.

Reply:
The point of this PR is to provide the **cause** of NoRollBackException, of course if it exists in the history.

Reply:
There are 5 causes for entries to be evicted, but we usually can't report why because we don't track cache eviction causes. If the history is big enough to capture when NoRollBackException, then the VLO can log exactly why the NoRollBackException happened.

Reply:
Yes, this I get. I was just wondering if we wanted to count the number of times we were replacing an entry in the cache, but now that it is not a loadingCache that actually shouldn't even be a removal cause in our case.